### ISSUE TITLE
Inconsistent use of FrozenBatchNorm in Faster-RCNN?

### ISSUE THREAD

**Issue by MoPl90** [pytorch/vision#6543]
Hi,
while customizing and training a Faster-RCNN object detection model based on `torchvision.models.detection.faster_rcnn`, I've noticed that the pre-trained model of type `fasterrcnn_resnet50_fpn_v2` always use `nn.BatchNorm2d` normalization layers, while `fasterrcnn_resnet50_fpn` uses `torchvision.models.ops.misc.FrozenBatchNorm2d` when pretrained weights are loaded. I've noticed deteriorating performance of the V2 model when training a COCO pretrained model with low batch size. I am suspecting that this is related to the un-frozen `nn.BatchNorm2d` layers, and indeed, replacing `nn.BatchNorm2d` with `torchvision.models.ops.misc.FrozenBatchNorm2d` improves the performance for my task. 

Thus, my question is: Is this discrepancy in normalization layers intentional, and if yes what could be other reasons for V2 model underperforming compared to the V1 model?

I'm using pytorch 1.12, torchvision 0.13.

Thanks!

cc @datumbox

**Comment by datumbox** [comment]
@MoPl90  Thanks for the question.

The change was intentional. The specific heuristic was previously used to assist with small batch size training. Unfortunately there was a bug on the logic on when the BN should be replaced, see #5443 for details. On V2, we got better results without freezing the BN and using syncBN instead, so we removed the heuristic.

**Comment by MoPl90** [comment]
Thanks for the quick reply. That's interesting, I'll have a closer look then.

**Comment by eflorico** [comment]
I have come across the same issue. Similar to @MoPl90, I have also found `fasterrcnn_resnet50_fpn_v2` to perform substantially worse than `fasterrcnn_resnet50_fpn` on a finetuning task, and I am wondering if the batchnorm layers might be the cause.

@datumbox I understand that BN layers were not frozen when training the `FasterRCNN_ResNet50_FPN_V2_Weights` checkpoint. But wouldn't it still make sense for downstream users of that checkpoint to freeze the batchnorm layers?

Currently, this is not possible (unless you re-write `fasterrcnn_resnet50_fpn_v2()`).
